index.js

```diff
@@ -75,7 +75,7 @@ let authS3O = function (req, res, next) {
 	} else {
 		const s3o_system_code = req.headers['x-s3o-systemcode'];
 		const protocol = (req.headers['x-forwarded-proto'] && req.headers['x-forwarded-proto'] === 'https') ? 'https' : req.protocol;
-		const originalLocation = protocol + '://' + req.headers.host + req.originalUrl;
+		const originalLocation = protocol + '://' + req.hostname + req.originalUrl;
```

Comment:
Would this be a breaking change?

Reply:
If an app is dependent on the unusual behaviour this results in when `trust proxy` is enabled, then I think so, e.g. `req.hostname == 'preview.ft.com'` while `req.http.host == 'ft-next-preview-eu.herokuapp.com'`.